mqtt-record/main.go

```diff
@@ -78,6 +98,46 @@ var message_handler mqtt.MessageHandler = func(client mqtt.Client, msg mqtt.Mess
 
 	file.Write(buf_size)
 	file.Write(buf_payload)
+
+	// calculate message statistics (omitting first message)
+	stats, exists := msgStats[topic]
+	if !exists {
+		var initStats = MsgStats{}
+		initStats.NumMsgs = 0
+		initStats.AvgMsgSizeByte = 0
+		initStats.AvgTimeDiffMillis = 0
+		initStats.LastMsgMillis = t
+
```

Comment:
The initial values for MinTimeDiffMillis etc. are missing. Does this causes the error on output?

Reply:
I'm refactoring that code currently. But somehow the average still goes to zero after some time. Min and max are fine.

Reply:
Probably integer divisions where floating point should have been.